how-it-works/00_General/01_overview-of-the-internet-computer.subpage.md

```diff
@@ -0,0 +1,125 @@
+
+<figure>
+<img src="/img/how-it-works/subnet_architecture.png" alt="Architecture: The IC is composed of subnets, each of which is an independent blockchain" title="The IC is composed of subnets, each of which is an independent blockchain" align="center" style="width:600px">
+<figcaption align="center">
+The IC is composed of subnets, each of which is an independent blockchain.<br>
+Canisters within and across subnets communicate through asynchronous messaging.
+</figcaption>
+</figure>
+
+## Canister Smart Contracts
+
+Each subnet can host tens of thousands of *canister smart contracs* or simply *canisters*.
+A canister smart contract is a bundle comprising the smart contract Wasm bytecode and its storage.
+Canister smart contracts can be updated by their maintainer, which can be a single developer, a groups of developers, or a decentralized autonomous organization (DAO).
+For reasons of decentralization, it is best if a canister smart contract be maintained by a DAO in the long run.
+
+Canisters pay for the IC resources they consume in *cycles*.
+To this end, canisters need to be topped up with cycles by their maintainer or anyone else who has an interest in the canister running.
+Cycles can be acquired with the ICP token, the IC's utility token.
+One Trillion cycles can be acquired with ICP worth 1 XDR, where an XDR is a basket comprising major currencies and one XDR is roughly 1.3 USD as of Q3 2022.
+
+Canisters can hold gigabytes of memory for a low fee.
+They can serve the user interface of a dApp directly to a web browser, unlike the UI being served from public cloud as is the case in other blockchains.
+Canisters can be updated and evolve, much like regular software.
+DAO-based governance can make such upgrade process secure and decentralized.
+Using Internet Identity, canisters can authenticate users based on biometry using the WebAuthn protocol.
+
+## Core Internet Computer Protocol
+
+Each subnet of the IC is driven by the core Internet Computer protocol (core IC protocol) an implementation of which is running on every node.
+This protocol consists of the following four layers: (1) The peer-to-peer, (2) consensus, (3) message routing, and (4) execution layer.
+This core IC protocol stack is running on all nodes of any subnet and drives the subnet to make progress in terms of consensus and message execution.
+Each subnet of the IC thereby is its own replicated state machine that makes progress independently of the other subnets of the IC.
+
+## Chain-Key Cryptography
+
+Many parts of the protocol of the IC depend on *chain-key cryptography*, also referred to as chain-key technology.
+Chain-key cryptography is the collection of cryptographic mechanisms that enable the Internet Computer Protocol.
+
+One important mechanism in this toolbox is non-interactive distributed key generation (NIDKG) for BLS signature keys.
```

Comment:
I think we need to start differently: before even explaining distributed key generation, we need to explain threshold cryptography, namely that a subnet together holds a cryptographic key. I'm also not sure it makes sense to mention BLS here, although that's at least a term that Ethereum users may know.